backend/src/routes/spotify.ts

```diff
@@ -72,13 +75,30 @@ router.get('/callback', async (ctx, next) => {
 
 	ctx.body = 'Success! You can now close the window.';
 
-	//Refresh the access token before it expires
+	// Refresh the access token before it expires.
 	setInterval(async () => {
+		console.log('This just exectued');
 		const data = await spotifyApi.refreshAccessToken();
 		const access_token = data.body['access_token'];
 
 		spotifyApi.setAccessToken(access_token);
 	}, (expires_in / 2) * 1000);
 });
 
+// Add a route for searching for a track on Spotify API.
+router.get('/search/:query', async (ctx, next) => {
+	const query = ctx.params.query;
+	var queryResult;
+	try {
+		queryResult = await spotifyApi.searchTracks(query);
+	} catch (error) {
+		console.log(`Received the following error from query: \n ${error}`);
+	}
+	const firstPage = queryResult.body.tracks.items;
```

Comment:
Could we replace the `.`s with `?.`s? As in, `queryResult?.body?.tracks?.items`. That way, if any of the fields don't exist, `firstPage` will just be `undefined`, rather than causing a runtime error. Then, we could check if `firstPage` is `undefined` and set the response to `[]` if so.